tensorflow/python/eager/function.py

```diff
@@ -3072,10 +3075,12 @@ def _cache_key(self, args, kwargs, include_tensor_ranks_only=False):
       inputs = (args, kwargs) if kwargs else args
       input_signature = pywrap_tfe.TFE_Py_EncodeArg(inputs,
                                                     include_tensor_ranks_only)
+      hashable_input_signature = _make_input_signature_hashable(input_signature)
     else:
       del args, kwargs
       assert not include_tensor_ranks_only
-      input_signature = self.flat_input_signature
+      assert hasattr(self, '_hashable_input_signature')
```

Comment:
why is this needed?

Reply:
Just to make sure the variable was initialized at object creation, since it only happens in the `input_signature is not None` case. In theory, I don't think the input_signature attribute changes after a Function object is initialized, and it doesn't look like we are doing any error handling if asserts fail, so I can remove if it feels extraneous. Thoughts?

Reply:
i think you can remove it, it is extra.

Reply:
Done